platform/logging/LoggerFactory.ts

```diff
@@ -0,0 +1,109 @@
+import { LoggingConfig } from './LoggingConfig';
+import { LogLevel } from './LogLevel';
+import { Logger, BaseLogger, LoggerAdapter } from './Logger';
+import { BaseAppender } from './appenders/base/BaseAppender';
+import { BaseAppenderConfig } from './appenders/base/BaseAppenderConfig';
+import { BufferAppender } from './appenders/buffer/BufferAppender';
+
+const BUFFERED_APPENDER_KEY = '__buffer__';
+
+export interface LoggerFactory {
+  get(...namespace: string[]): Logger;
+}
+
+export class MutableLoggerFactory implements LoggerFactory {
+  private config?: LoggingConfig;
+  private readonly appenders: Map<string, BaseAppender> = new Map([
```

Comment:
That the initial idea on how to handle log _before_ config is loaded, works well for now.